test/integration_test/fafb_matrix_test.go

```diff
@@ -6,15 +6,26 @@ import (
 	"testing"
 	"time"
 
+	"github.com/sirupsen/logrus"
```

Comment:
Can we rename this file to just `fafb_test.go`? I think it was just named *matrix* because it Adam had a matrix of +ve and -ve test cases which he was looping through.